playback/src/player.rs

```diff
@@ -651,18 +651,14 @@ impl PlayerTrackLoader {
                 FileFormat::OGG_VORBIS_96,
             ],
         };
-        let format = formats
-            .iter()
-            .find(|format| audio.files.contains_key(format))
-            .unwrap();
 
-        let file_id = match audio.files.get(&format) {
```

Comment:
I think the previous implementation was possibly clearer to read.

Reply:
But it was strange: In the first statement we get the key from the map, and in the second we get the item, which could be done in one operation. Besides that, the `audio.files.get` could never fail (or am I wrong?) but the first statement could in theory.

I will try to think of another way to express this.

Reply:
I can't think of a clearer way to express this, except of desugaring the question mark operator I used, so leave it as is? Idk.

Reply:
I'd probably leave as is. With regards to `audio.files.get`, from reading over the code it looks as though it might be possible for it to fail in the event that the list of alternatives contains > 0 alternatives, but none of them are available for playback.

Reply:
As far as I understood it, `audio.files` is a simple [`LinearMap`](https://docs.rs/linear-map/1.2.0/linear_map/struct.LinearMap.html). By using `find` and `unwrap`, we know that `audio.files.contains_key(&format) == true`, so `audio.files.get(&format)` should never be `None`.

Reply:
I think commit [`8eb51e9`](https://github.com/librespot-org/librespot/commit/8eb51e9b55d6051ced1275ac15a5578b307a3eb0#diff-1b550ae8c2dfcb42fe4328abb560702e7345603215fff908c0b5a227abff0fc6L548-L589) explains why this is implemented this way: Before this commit `format` was fixed, and we simply try to get that format from the map, and if it's not present, it fails. After this commit, `format` is determined as the first element of a list of formats that is also present in the map, and therefore this check became unnecessary, but wasn't removed.

Reply:
Is it now better?

Reply:
Yeah that looks fine.